test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/ClientYamlTestSuite.java

```diff
@@ -73,9 +74,10 @@ public static ClientYamlTestSuite parse(NamedXContentRegistry executeableSection
     }
 
     public static ClientYamlTestSuite parse(String api, String suiteName, XContentParser parser) throws IOException {
-        parser.nextToken();
-        assert parser.currentToken() == XContentParser.Token.START_OBJECT : "expected token to be START_OBJECT but was "
-                + parser.currentToken();
+        if (parser.nextToken() != XContentParser.Token.START_OBJECT) {
+            throw new XContentParseException(parser.getTokenLocation(),
+                    "expected token to be START_OBJECT but was " + parser.currentToken());
+        }
```

Comment:
I hit this assertion during development, which was not nice because it doesn't tell the line/offset of the problem, so I switched to a hard check that also gives the location of the parsing error.

Reply:
++